### ISSUE TITLE
Failure after block height 50500 while importing bootstrap.dat

### ISSUE THREAD

**Issue by Gnos1s** [Anoncoin/oldrepo-backup#100]
This is with the latest in the gr_develop branch (commit 44053a1). It appears that the failure happens just after the transition from the "old algo" to KGW. We should look in the ContextualCheckBlockHeader function.

```
2016-03-22 06:42:56 RetargetPID NOTE: OutputTime 1.000000 was < 1 second, out-of-range value set to 1.
2016-03-22 06:42:56 RetargetPID NOTE: Difficulty 1c02e2c0 was out of range and set to limit 1d01868d
2016-03-22 06:42:56 RetargetPID charged to height=50495, tipfilter updated, Next ProofOfWork based on old algo. Required=0x1d02076f
2016-03-22 06:42:56 RetargetPID charged to height=50495 output terms P=-403.096774 I=160.000000 D=-0.000000, ProofOfWork Required=0x1d01868d Header=0x1d02076f
2016-03-22 06:42:56 RetargetPID NOTE: OutputTime 1.000000 was < 1 second, out-of-range value set to 1.
2016-03-22 06:42:56 RetargetPID NOTE: Difficulty 1c02e2c0 was out of range and set to limit 1d01868d
2016-03-22 06:42:56 RetargetPID charged to height=50496, tipfilter updated, Next ProofOfWork based on old algo. Required=0x1d02076f
2016-03-22 06:42:56 RetargetPID charged to height=50496 output terms P=-404.354839 I=160.000000 D=-0.000000, ProofOfWork Required=0x1d01868d Header=0x1d02076f
2016-03-22 06:42:56 RetargetPID NOTE: OutputTime 1.000000 was < 1 second, out-of-range value set to 1.
2016-03-22 06:42:56 RetargetPID NOTE: Difficulty 1c02e2c0 was out of range and set to limit 1d01868d
2016-03-22 06:42:56 RetargetPID charged to height=50497, tipfilter updated, Next ProofOfWork based on old algo. Required=0x1d02076f
2016-03-22 06:42:56 RetargetPID charged to height=50497 output terms P=-406.403226 I=160.000000 D=-0.000000, ProofOfWork Required=0x1d01868d Header=0x1d02076f
2016-03-22 06:42:56 RetargetPID NOTE: OutputTime 1.000000 was < 1 second, out-of-range value set to 1.
2016-03-22 06:42:56 RetargetPID NOTE: Difficulty 1c02e2c0 was out of range and set to limit 1d01868d
2016-03-22 06:42:56 RetargetPID charged to height=50498, tipfilter updated, Next ProofOfWork based on old algo. Required=0x1d02076f
2016-03-22 06:42:56 RetargetPID charged to height=50498 output terms P=-408.440860 I=160.000000 D=-0.000000, ProofOfWork Required=0x1d01868d Header=0x1d02076f
2016-03-22 06:42:56 RetargetPID NOTE: OutputTime 1.000000 was < 1 second, out-of-range value set to 1.
2016-03-22 06:42:56 RetargetPID NOTE: Difficulty 1c02e2c0 was out of range and set to limit 1d01868d
2016-03-22 06:42:56 RetargetPID charged to height=50499, tipfilter updated, Next ProofOfWork based on old algo. Required=0x1d02076f
2016-03-22 06:42:56 RetargetPID charged to height=50499 output terms P=-406.736559 I=160.000000 D=0.000000, ProofOfWork Required=0x1d01868d Header=0x1d02076f
2016-03-22 06:42:56 RetargetPID NOTE: OutputTime 1.000000 was < 1 second, out-of-range value set to 1.
2016-03-22 06:42:56 RetargetPID NOTE: Difficulty 1c02e2c0 was out of range and set to limit 1d01868d
2016-03-22 06:42:56 RetargetPID charged to height=50500, tipfilter updated, Last Block based on KGW. ProofOfWork Required=0x1d02076f
2016-03-22 06:42:56 ERROR: ContextualCheckBlockHeader : rejected nVersion=1 block
2016-03-22 06:42:56 ERROR: ProcessNewBlock : AcceptBlock FAILED
2016-03-22 06:42:56 LoadExternalBlockFile : Out of order block 0000000063adb9e52a5994041280e1004803bc2141b82d7f7ab9e044fe304586, parent 0000000000000000000000000000000000000000000000000000000000000000 not known
2016-03-22 06:42:56 LoadExternalBlockFile : Out of order block 00000001a0b1ab06f83b4a680bf3575877eaf714ba950b6a260c652852cd9519, parent 0000000000000000000000000000000000000000000000000000000000000000 not known
2016-03-22 06:42:56 LoadExternalBlockFile : Out of order block 000000004e426198fd52e1d4609da75077169c5a2e8fb331e90cd04a76118d33, parent 0000000000000000000000000000000000000000000000000000000000000000 not known
2016-03-22 06:42:56 LoadExternalBlockFile : Out of order block 000000014fc64caf31466bbedf5fdb94f5c18406205fcccaecf63ec6e75ff9b0, parent 0000000000000000000000000000000000000000000000000000000000000000 not known
2016-03-22 06:42:56 LoadExternalBlockFile : Out of order block 00000001ef9cc71f2cc0b7b75317830205130fec689325587c2d4afd8ac3ecda, parent 0000000000000000000000000000000000000000000000000000000000000000 not known
2016-03-22 06:42:56 LoadExternalBlockFile : Out of order block 00000000fcf882f4d806f1fc9115632e5eac0d1d42cc63aa66befaf017021be8, parent 0000000000000000000000000000000000000000000000000000000000000000 not known
2016-03-22 06:42:56 LoadExternalBlockFile : Out of order block 00000000da90b70729bcc2d5fb6d3396d91a3c51d66f2b23a30ab872cad29390, parent 0000000000000000000000000000000000000000000000000000000000000000 not known
2016-03-22 06:42:56 LoadExternalBlockFile : Out of order block 00000000b33da8b175fc4788ec6c41eb7b7bcf2c0f27cac1365f58172b5a3a6b, parent 0000000000000000000000000000000000000000000000000000000000000000 not known
2016-03-22 06:42:56 LoadExternalBlockFile : Out of order block 00000001743cdecb958f44461b15feca210bc495cee56ae861c7fb3d856d8cdc, parent 0000000000000000000000000000000000000000000000000000000000000000 not known
```

**Comment by Cryptoslave** [comment]
I do not have this issue in Antergos synching from bootstrap.dat using anoncoind or anoncoin-qtc.  I also am not able to reproduce it in win64 synching from scratch through i2p. I also have much less debug than you, only once every 1000 blocks. You shall synch the bootstrap.dat without any debug option enabled otherwise it is too much slow. Check this anoncoin.conf: https://github.com/Anoncoin/anoncoin/blob/gr_develop/doc/anoncoin.conf.sample

Can you retry it again with this particular bootstrap that I used and no debug? 
https://anoncoin.net/downloads/bootstrap/

**Comment by Gnos1s** [comment]
This was me being dumb. I made a typo in the configure command and had --with-hardfork=50500 instead of 505000.

**Comment by ssua6** [comment]
Reproduces on my machine where I get stuck at block 504122 even after using that fresh bootstrap and same anoncoin.conf.

```
2016-03-26 00:37:53 CService(5oo3enrz7fp77ojrfk7hjsniohsxqmhuxdhdx6ur7iwumsrjzkwq.b32.i2p)
2016-03-26 00:37:53 CService(xowpui5nxkarsg2uwjllc6wdteheytknicbbqsnbkjjwde5iq6ma.b32.i2p)
2016-03-26 00:38:16 ERROR: ContextualCheckBlockHeader : incorrect proof of work
2016-03-26 00:38:16 ERROR: invalid header received
2016-03-26 00:38:16 ProcessMessage(headers, 162003 bytes) FAILED, from peer=36
2016-03-26 00:38:28 accepted connection yg5zv3anh63l22pq3vvb2znaubpl7bbq4lhlhyfvzkmddunmuuya.b32.i2p
2016-03-26 00:38:40 receive version message: /Anonymous:0.9.6.10/: version 70010, blocks=507212, services=0x0000000000000081 us=btc4x6eixkhsnfj4gdquky4qsx
64tu53tofseb6a6zflgfpzo4ta.b32.i2p, them=yg5zv3anh63l22pq3vvb2znaubpl7bbq4lhlhyfvzkmddunmuuya.b32.i2p, peer=yg5zv3anh63l22pq3vvb2znaubpl7bbq4lhlhyfvzkmddunmuuya.b32.i2p
2016-03-26 00:38:42 CService(xowpui5nxkarsg2uwjllc6wdteheytknicbbqsnbkjjwde5iq6ma.b32.i2p)
2016-03-26 00:38:42 CService(b7ziruwpk7g2e44xyomnc2nu5tx7bc2f2ai4dzi66uxm3bc3qttq.b32.i2p)
```

**Comment by Gnos1s** [comment]
@ssua6 that is a different error. Yours says "ContextualCheckBlockHeader : incorrect proof of work" whereas mine says "ContextualCheckBlockHeader : rejected nVersion=1 block". Take a look at main.cpp near line 2730. :+1:

**Comment by Cryptoslave** [comment]
Also block 504122 is not in the bootstrap. The bootstrap go to around February 2015 so around block 315000. Then the client sync using the connected peers.

You shall tell us who is peer=36 and its version. Normally it cannot do the initial block download from a peer before protocol 70010 at this point in time (from block 502600 onward).

Also like usual, getchaintips is your friend here, check was it say after block 500000.

If you did the IBD from a 70009 peer before block 502600 some strange things may happen, but normally not before 505001.

Give more detail we cannot help you without getchaintips and detail on peer=36.

**Comment by ssua6** [comment]
This issue under the ticket title has been resolved. Opening new issue to track my comment above.
